gaurav/src/pages/recipe/SearchResults.tsx

```diff
@@ -0,0 +1,173 @@
+  const [recipesPerPage, setRecipesPerPage] = useState(6);
+
+  const query = new URLSearchParams(location.search).get("query");
+
+  const fetchRecipes = async (query = ""): Promise<IData[]> => {
+    try {
+      const res = await axios.get(`https://www.themealdb.com/api/json/v1/1/search.php?s=${query}`);
+      const data = res.data;
+      setSearchResults(data.meals || []);
+      return data.meals || [];
+    } catch (error) {
+      handleShowToast("No Recipe Found");
+      return [];
+    }
+  };
+
+  useEffect(() => {
+    fetchRecipes(query || "");
+  }, [query]);
+
+  const handleSearch = async (e: React.FormEvent) => {
+    e.preventDefault();
+    const result = await fetchRecipes(searchQuery);
+    setSearchResults(result);
+    navigate(`/searchresults?query=${searchQuery}`);
+  };
+
+  const handleSavedRecipe = (id: string): void => {
+    const recipe = searchresults.find((recipe) => recipe.idMeal === id);
+    if (recipe) {
+      const isAlreadySaved = savedRecipe.some((saved) => saved.idMeal === id);
+      if (!isAlreadySaved) {
+        const updatedSavedRecipes = [...savedRecipe, recipe];
+        setSavedRecipe(updatedSavedRecipes);
+        localStorage.setItem("savedRecipe", JSON.stringify(updatedSavedRecipes));
+        handleShowToast("Recipe Saved Successfully");
+        setSavedStatus((prevStatus) => ({ ...prevStatus, [id]: true }));
+        localStorage.setItem("savedStatus", JSON.stringify({ ...savedStatus, [id]: true }));
+      } else {
+        handleShowToast("Recipe Already Saved");
+      }
+    }
+  };
+
+  const handleRemoveRecipe = (id: string): void => {
+    const updatedRecipe = savedRecipe.filter((recipe) => recipe.idMeal !== id);
+    setSavedRecipe(updatedRecipe);
+    localStorage.setItem("savedRecipe", JSON.stringify(updatedRecipe));
+    handleShowToast("Recipe Removed Successfully");
+    setSavedStatus((prevStatus) => ({ ...prevStatus, [id]: false }));
+    localStorage.setItem("savedStatus", JSON.stringify({ ...savedStatus, [id]: false }));
+  };
+
+  const handleClick = (id: string) => {
+    navigate(`/recipedetail/${id}`);
+  };
+
+  useEffect(() => {
+    const savedRecipe = localStorage.getItem("savedRecipe");
+    if (savedRecipe) {
+      setSavedRecipe(JSON.parse(savedRecipe));
+    }
+  }, []);
+
+  useEffect(() => {
+    const savedRecipeStatus = localStorage.getItem("savedStatus");
+    if (savedRecipeStatus) {
+      setSavedStatus(JSON.parse(savedRecipeStatus));
+    }
+  }, []);
+
+  const handleShowToast = (message: string) => {
+    setToastMessage(message);
+    setShowToast(true);
+    setTimeout(() => {
+      setShowToast(false);
+    }, 3000);
+  };
+
+  // Pagination Logic
+  const indexOfLastRecipe = currentPage * recipesPerPage;
+  const indexOfFirstRecipe = indexOfLastRecipe - recipesPerPage;
+  const currentRecipes = searchresults.slice(indexOfFirstRecipe, indexOfLastRecipe);
+
+  return (
+    <>
+      <div className="bg-[#083344] min-h-screen relative">
```

Comment:
bg-[#083344] - adding color codes like this is bad practice. Use CSS variable names